src/wallet/hdchain.h

```diff
@@ -0,0 +1,70 @@
+#ifndef ZCOIN_HDCHAIN_H
+#define ZCOIN_HDCHAIN_H
+
+#include "key.h"
+
+/* simple HD chain data model */
+class CHDChain
```

Comment:
You absolutely cannot move CHDChain to another file because it's bitcoin code. It will create a lot of problems when upgrading to newer bitcoin cores

Reply:
In Dash it is done the same way.

Reply:
not a good example because they implemented HD stuff to 0.12 version of their bitcoin core, we're already at 0.13 and should not mess with that

Reply:
ok, good, will move it back.